lib/irb/cmd/history.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "stringio"
+require_relative "nop"
+require_relative "../pager"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class History < Nop
+      category "IRB"
+      description "Shows the input history. `-g [query]` or `-G [query]` allows you to filter the output."
+
+      def self.transform_args(args)
+        if match = args&.match(/\A(?<args>.+\s|)(-g|-G)\s+(?<grep>[^\s]+)\s*\n\z/)
```

Comment:
I think we can accept `\s` in regexp. `history -g if match` will be useful. Pry's `hist --grep` accepts spaces.

Reply:
Good idea!

- Allowing whitespace: https://github.com/ruby/irb/pull/761/commits/c48b70c3c07d184aac8e94394c69621bda23054c
- Removing `args`: https://github.com/ruby/irb/pull/761/commits/6ebd08c142e01c00f5cea67201e1b1c4a4e5da90